client/my-sites/subscribers/components/subscriber-data-views/subscriber-data-views.tsx

```diff
@@ -281,6 +264,35 @@ const SubscriberDataViews = ( {
 		};
 	}, [ subscribers, grandTotal, pages ] );
 
+	// Update the view when a subscriber is selected
+	useEffect( () => {
+		const commonViewProps = {
+			page,
+			perPage,
+			sort: {
+				field: sortTerm,
+				direction: sortOrder,
+			},
+		};
+
+		setCurrentView( ( oldCurrentView ) => ( {
+			...oldCurrentView,
+			...commonViewProps,
+			type: selectedSubscriber ? 'list' : 'table',
+			layout: selectedSubscriber
+				? {
+						showMedia: true,
+						mediaSize: 40,
+						mediaField: 'media',
+						primaryField: 'name',
+				  }
+				: {},
+			fields: selectedSubscriber
+				? [ 'media', 'name' ]
+				: [ 'name', ...( ! isMobile ? [ 'subscription_type', 'date_subscribed' ] : [] ) ],
+		} ) );
+	}, [ isMobile, selectedSubscriber, page, perPage, sortTerm, sortOrder ] );
```

Comment:
Centralizing the view state made it easier to read and understand what was changing.